### ISSUE TITLE
compute_meandice calculates nan for full zero input

### ISSUE THREAD

**Issue by iwan933** [Project-MONAI/MONAI#2266]
**Describe the bug**
The function `compute_meandice` outputs nan if labels for a channel are all 0 and predictions are 0 as well. (Can happen while cropping e.g.)

**To Reproduce**
```
>>> from monai.metrics import compute_meandice
>>> import torch
>>> pred = torch.zeros((1, 2, 5, 5))
>>> labels = torch.zeros((1, 2, 5, 5))
>>> compute_meandice(y_pred=pred, y=labels, include_background=False)
tensor([[nan]])
```

**Expected behavior**
Return value should equal `1.0`.

**Environment**

```
MONAI version: 0.5.2
Numpy version: 1.20.2
Pytorch version: 1.8.1+cpu
MONAI flags: HAS_EXT = False, USE_COMPILED = False
MONAI rev id: feb3a334b7bbf302b13a6da80e0b022a4cf75a4e

Optional dependencies:
Pytorch Ignite version: NOT INSTALLED or UNKNOWN VERSION.
Nibabel version: 3.2.1
scikit-image version: NOT INSTALLED or UNKNOWN VERSION.
Pillow version: 8.2.0
Tensorboard version: 2.4.1
gdown version: NOT INSTALLED or UNKNOWN VERSION.
TorchVision version: 0.9.1+cpu
ITK version: NOT INSTALLED or UNKNOWN VERSION.
tqdm version: 4.60.0
lmdb version: NOT INSTALLED or UNKNOWN VERSION.
psutil version: NOT INSTALLED or UNKNOWN VERSION.

For details about installing the optional dependencies, please visit:
    https://docs.monai.io/en/latest/installation.html#installing-the-recommended-dependencies


================================
Printing system config...
================================
`psutil` required for `print_system_info`

================================
Printing GPU config...
================================
Num GPUs: 0
Has CUDA: False
cuDNN enabled: False
```

**Solution**
Catch the case by checking the denominator value for 0. I can create a pullrequest if accepted.

https://github.com/Project-MONAI/MONAI/blob/12bbe776735ba05b68cde020cfface8038915f14/monai/metrics/meandice.py#L125

**Comment by Nic-Ma** [comment]
Hi @iwan933 ,

Thanks for your interest and feedback.
I think Nan is expected result for all zero data, @ericspod @wyli  could you please help double confirm?

Thanks.

**Comment by wyli** [comment]
yes there are discussions https://github.com/Project-MONAI/MONAI/issues/887, I'm closing this
